File: backend/app/schemas/page.py
from typing import Any, Literal

from pydantic import BaseModel, Field

from app.schemas.component import DesignComponent


class StoreDesignPage(BaseModel):
    id: int | str = ""
    diy_id: int | str = ""
    company_id: int | str | None = None
    type: str = "home_page"
    page_name: str
    datas: list[DesignComponent] = Field(default_factory=list)
    other_company_show: int = 0
    page_info: dict[str, Any] = Field(default_factory=dict)
    member_level: list[Any] = Field(default_factory=list)
    level: list[Any] = Field(default_factory=list)
    status: int = 1
    created_at: str | None = None
    updated_at: str | None = None
    page_sort: int | str = 2
    page_scene: int | str = 2
    top_id: dict[str, Any] = Field(default_factory=dict)
    foot_type: int | str = 1
    foot_id: dict[str, Any] = Field(default_factory=dict)
    page_type: str | list[str] = "2"


class SavePagePayload(StoreDesignPage):
    mode: Literal["create", "edit", "copy"] = "create"
    route_id: int | str | None = None
    head_img: str | None = None


class SavePageResult(BaseModel):
    id: str
    diy_id: str
    url: str
    message: str

File: frontend/src/domain/page.ts
import type { DesignComponent } from './component'

export type StoreDesignPage = {
  id: number | string
  diy_id: number | string
  company_id?: number | string
  type: 'home_page' | 'project_page' | string
  page_name: string
  datas: DesignComponent[]
  other_company_show?: 0 | 1
  page_info: Record<string, unknown>
  member_level?: number[] | string | null
  level: unknown[]
  status?: number
  created_at?: string
  updated_at?: string
  page_sort?: number | string
  page_scene?: number | string
  top_id?: Record<string, unknown> | null
  foot_type?: number | string
  foot_id?: Record<string, unknown> | null
  page_type?: string | string[]
}

export type SavePagePayload = StoreDesignPage & {
  mode: 'create' | 'edit' | 'copy'
  route_id?: string | number
  head_img?: string
}

export type SavePageResult = {
  id: string
  diy_id: string
  url: string
  message: string
}
